libHVCAENx527App/src/CAENHVWrapper_sim.cpp

```diff
@@ -60,6 +66,10 @@ CAENHVLIB_API CAENHVRESULT  CAENHVGetChName(const char *SystemName, ushort slot,
 CAENHVLIB_API CAENHVRESULT  CAENHVSetChName(const char *SystemName, ushort slot, 
  ushort ChNum, const ushort *ChList, const char *ChName)
 {
+    if (++g_call_out % g_error_on_call == 0) {
+	    return CAENHV_TIMEERR;
+	}	
```

Comment:
This logic is repeated in multiple places. Refactor to eliminate duplication. 

Also see comment above, I'm not sure this is the best approach.